lib/spack/spack/relocate.py

```diff
@@ -0,0 +1,273 @@
+    The old install dir in LC_LOAD_DYLIB is replaced with the new install dir
+    using install_name_tool -change old new binary
+    The old install dir in LC_RPATH is replaced with the new install dir using
+    install_name_tool  -rpath old new binary
+    """
+    # avoid error message for libgcc_s
+    if 'libgcc_' in path_name:
+        return
+    rpaths, deps, idpath = macho_get_paths(path_name)
+    id = None
+    nrpaths = []
+    ndeps = []
+    if relative:
+        nrpaths, ndeps, id = macho_make_paths_relative(path_name,
+                                                       old_dir, rpaths,
+                                                       deps, idpath)
+    else:
+        nrpaths, ndeps, id = macho_replace_paths(old_dir, new_dir, rpaths,
+                                                 deps, idpath)
+    st = os.stat(path_name)
+    # some installs create read-only binaries
+    install_name_tool = Executable('install_name_tool')
+    if id:
+        install_name_tool('-id', id, path_name, output=str, err=str)
+
+    for orig, new in zip(deps, ndeps):
+        install_name_tool('-change', orig, new, path_name)
+
+    for orig, new in zip(rpaths, nrpaths):
+        install_name_tool('-rpath', orig, new, path_name)
+        os.chmod(path_name, st.st_mode)
+    return
+
+
+def get_filetype(path_name):
+    """
+    Check the output of the file command for given string.
+    """
+    file = Executable('file')
+    file.add_default_env('LC_ALL', 'C')
+    output = file('-b', '-h', '%s' % path_name,
+                  output=str, err=str)
+    return output.strip()
+
+
+def modify_elf_object(path_name, orig_rpath, new_rpath):
+    """
+    Replace RPATH's in given elf object
+    """
+    st = os.stat(path_name)
+    wmode = os.access(path_name, os.W_OK)
+    if not wmode:
+        os.chmod(path_name, st.st_mode | stat.S_IWUSR)
+    if platform.system() == 'Linux':
+        new_joined = ':'.join(new_rpath)
+        patchelf = Executable(get_patchelf())
+        patchelf('--force-rpath', '--set-rpath', '%s' % new_joined,
+                 '%s' % path_name, output=str, cmd=str)
+    else:
+        tty.die('relocation not supported for this platform')
+    os.chmod(path_name, st.st_mode)
+
+
+def needs_binary_relocation(filetype):
+    """
+    Check whether the given filetype is a binary that may need relocation.
+    """
+    retval = False
+    if "relocatable" in filetype:
+        return False
+    if platform.system() == 'Darwin':
+        return ('Mach-O' in filetype)
+    elif platform.system() == 'Linux':
+        return ('ELF' in filetype)
+    else:
+        tty.die("Relocation not implemented for %s" % platform.system())
+    return retval
+
+
+def needs_text_relocation(filetype):
+    """
+    Check whether the given filetype needs relocation.
+    """
+    return ("text" in filetype)
+
+
+def relocate_binary(path_name, old_dir, new_dir):
+    """
+    Change RPATHs in given elf or mach-o file
```

Comment:
Could you update these comments to refer to the parameter names? For example: "Change RPATHs in the file located at path_name by replacing old_dir with new_dir"